### PR TITLE
Update plugin to support ASU Directory and add new action hooks

### PR DESCRIPTION
**Pull request by ndrollin**
Sorry, there is probably more in here than should be in a single PR. The main focus was to add support for ASURITE users. This involves adding support for pulling user info from the ASU Directory, and revamping the language throughout the plugin, particularly in the settings views, to make it clearer how to use the plugin.

New action hooks were also added and described in the README. Most other changes to the plugin are minor and were the result of reviewing the plugin logic and debugging the behavior in the user registration process.

### CODE REVIEW

**Comment by ndrollin** [comment]
I just remembered I wanted to run this plugin through phpcs and phpcbf to clean up for coding standards.

File: cas-maestro.php [hunk 1]
```diff
@@ -16,308 +17,356 @@
...
-    //Initialize the settings
+    // Initialize the settings
     $default_settings = array(
-        'cas_menu_location'=>'sidebar',
+        'cas_menu_location' => 'settings',
         'new_user' => false,
           'email_suffix' => '',
-          'cas_version' => "1.0",
+          'cas_version' => '1.0',
           'server_hostname' => 'yourschool.edu',
```
Comment (rraub) [inline cas-maestro.php]: Since this is the "ASU" Version can we set this to 'weblogin.asu.edu' ?

File: cas-maestro.php [hunk 2]
```diff
@@ -16,308 +17,356 @@
...
           'server_port' => '443',
           'server_path' => '',
```
Comment (rraub) [inline cas-maestro.php]: also can we add `cas` to the default server path?

File: cas-maestro.php [hunk 3]
```diff
@@ -16,308 +17,356 @@
...
-        add_filter('login_url', array(&$this, 'bypass_reauth'));
-        add_action('lost_password', array(&$this, 'disable_function'));
-        add_action('retrieve_password', array(&$this, 'disable_function'));
-        add_action('password_reset', array(&$this, 'disable_function'));
-        add_filter('show_password_fields', array(&$this, 'show_password_fields'));
+        add_filter( 'authenticate', array( &$this, 'validate_login' ), 30, 3 );
+        add_filter( 'login_url', array( &$this, 'bypass_reauth' ) );
+        // add_action('lost_password', array(&$this, 'disable_function'));
+        // add_action('retrieve_password', array(&$this, 'disable_function'));
+        // add_action('password_reset', array(&$this, 'disable_function'));
```
Comment (rraub) [inline cas-maestro.php]: Why did these actions get removed?

File: cas-maestro.php [hunk 4]
```diff
@@ -16,308 +17,356 @@
...
-     if (function_exists('load_plugin_textdomain')) {
-         load_plugin_textdomain( 'CAS_Maestro', false, dirname( plugin_basename( __FILE__ ) ) . '/lang/');
-     }
+    if ( function_exists( 'load_plugin_textdomain' ) ) {
+      load_plugin_textdomain( 'CAS_Maestro', false, dirname( plugin_basename( __FILE__ ) ) . '/lang/' );
+    }
   }
 
   /**
    * Generates a random string with a lenght
```
Comment (rraub) [inline cas-maestro.php]: typo: leng**T**h

File: cas-maestro.php [hunk 5]
```diff
@@ -16,308 +17,356 @@
...
-    if(!is_wp_error($user)) {
-      if(!isset($_SESSION))
+  /*
+  ----------------------------------------------*
+  * Authentication managment
+  *----------------------------------------------
+  */
+
+  function validate_noncas_login( $user, $username, $password ) {
+    $this->_write_log( __CLASS__ . '.' . __FUNCTION__ . ' (' . __LINE__ . ') : ' );
```
Comment (rraub) [inline cas-maestro.php:254]: are all these log messages just for debugging?

File: vendors/asurite/asu_directory.php [hunk 6]
```diff
@@ -0,0 +1,124 @@
+<?php // this is sourced from the GIOS API
+// This is a static class for interacting with the ASU public directory
+class ASUDirectory {
```
Comment (rraub) [inline vendors/asurite/asu_directory.php:3]: This being copy/pasted in isn't my favorite way to do this.. but adding composer just for this file is a bit of work too.  The only reason I bring this up is that these directory endpoints will be going away "soon" and we'll need to figure out equivalent ones using the new iSearch tools.

**Review by rraub — CHANGES_REQUESTED** [review]

**Comment by rraub** [comment]
also we should make some changes to the header of cas-maestro.php so that our plugin wont get updated and overridden if there is ever another update from the official source for CAS Maestro.

File: cas-maestro.php (hunk 3 above)
Comment (ndrollin) [reply cas-maestro.php]: Oops. I jumped the gun on issuing this PR. Those are just temporarily disabled until I work out a minor routing issue with non-CAS users. I described it in the specific commit, but obviously should have also added a TODO/FIXME comment.

File: cas-maestro.php (hunk 4 above)
Comment (ndrollin) [reply cas-maestro.php]: I don't understand what you are pointing out.

File: cas-maestro.php (hunk 5 above)
Comment (ndrollin) [reply cas-maestro.php:254]: Yeah, they only run when WP_DEBUG == TRUE. I was thinking I might remove them eventually.

File: cas-maestro.php (hunk 4 above)
Comment (rraub) [reply cas-maestro.php]: the comment says "lenght" not "length", its not worth digging in just to fix it, but if you already are working on that file then its a quick fix. Chances are its spelled wrong in the original plugin.

Comment (ndrollin) [reply cas-maestro.php]: Oh, hah. I was focused on the edited part of that preview.

**Review by rraub — APPROVED** [review]

**Pull request merged**